syntax = "proto2";

package aibench;

option optimize_for = LITE_RUNTIME;

enum BenchmarkOption {
  Performance = 1;
  Precision = 2;
}

enum ExecutorType {
  MACE = 1;
  SNPE = 2;
  TFLITE = 3;
  NCNN = 4;
  HIAI = 5;
  MNN = 6;
  TNN = 7;
}

enum DeviceType {
  CPU = 1;
  GPU = 2;
  DSP = 3;
  NPU = 4;
  VULKAN = 5;
  OPENGL = 6;
}

enum ModelName {
  MobileNetV1 = 1;
  MobileNetV2 = 2;
  InceptionV3 = 3;
  SqueezeNetV11 = 4;
}

enum ChannelOrder {
  RGB = 1;
  BGR = 2;
}

enum DataFormat {
  NHWC = 1;
  NCHW = 2;
}

message PreProcessor {
  enum PreProcessorType {
    DefaultProcessor = 1;
  }
  optional PreProcessorType type = 1;

  message MeanValue {
    repeated float mean = 1;
  }
  repeated MeanValue input_mean = 2;
  repeated float var = 3;
}

message PostProcessor {
  enum PostProcessorType {
    ImageClassification = 1;
    SSDDetection = 2;
    YOLOV3Detection = 3;
  }
  optional PostProcessorType type = 1;
  optional float threashold = 2 [default = 0.5];
}

message MetricEvaluator {
  enum MetricEvaluatorType {
    ImageClassification = 1;
    COCOObjectDetection = 2;
  }
  optional MetricEvaluatorType type = 1;
}
